conan/internal/graph/graph_binaries.py

```diff
@@ -448,8 +449,20 @@ def _evaluate_single(n):
             for node in level:
                 nodes.setdefault(node.pref, []).append(node)
             # PARALLEL, this is the slow part that can query servers for packages, and compatibility
-            for pref, pref_nodes in nodes.items():
-                _evaluate_single(pref_nodes[0])
+            host_nodes = [pref_nodes[0] for pref_nodes in nodes.values()
```

Comment:
Not very clear why this division.
why the separation of host and build? Why not call them "build" nodes? Why checking first the "host" nodes?